### ISSUE TITLE
when use --load to load quantized model,  the quantized weight is not correct ??

### ISSUE THREAD

**Issue by HueCheng1021** [Vahe1994/SpQR#48]
in spqr_engine.py:
        if save_quantization:
            save_quant_dict["perm"] = perm.to(torch.int32)
            save_quant_dict["keep_last_columns"] = 0
            save_quant_dict["blocksize"] = 128
            save_quant_dict["weight_shape"] = weight.shape
            save_quant_dict["groupsize"] = groupsize if groupsize else weight.shape[1]
            save_quant_dict["quant_weights"] = torch.cat(save_quant_dict["quant_weights"], dim=1)
            save_quant_dict["outliers_matrix"] = save_quant_dict["outliers_matrix"].to_sparse()
            **save_quant_dict["final_weight"] = weight** # for debug

Test demo:
model_path = 'pretrain_models/Llama-2-7b-hf'
load_path = 'SpQR/output_model_new'
model = get_model(model_path, None).train(False)
layers = get_layers(model)
for i in trange(len(layers)):
    layer = layers[i]
    sub_layers = find_sublayers(layer)
    for name in sub_layers:
        quantized_params_dict = read_quant_weight_from_file(load_path, i, name)
        dequant_weight = quantized_params_dict["final_weight"].cuda() 
        reconstructed_weight = layer_weight_dequantization(quantized_params_dict).cuda()
        print('dequant_weight:{}'.format(dequant_weight))
        print('reconstructed_weight:{}'.format(reconstructed_weight))
        print((dequant_weight - reconstructed_weight).abs().max())
        print(((dequant_weight - reconstructed_weight).abs() == (dequant_weight - reconstructed_weight).abs().max()).nonzero())
        print(dequant_weight[2074][1544], reconstructed_weight[2074][1544])
        print(dequant_weight[2074][1710], reconstructed_weight[2074][1710])
        print(dequant_weight[2078][3238], reconstructed_weight[2078][3238])
        exit(-1)
###############PRINT######################
Loading pretrained model ...
Loading checkpoint shards: 100%|██████████| 2/2 [00:01<00:00,  1.43it/s]
Model loaded sucessfully ...
  0%|          | 0/32 [00:00<?, ?it/s]dequant_weight:tensor([[-0.0041, -0.0151, -0.0017,  ...,  0.0025,  0.0024, -0.0037],
        [ 0.0178, -0.0057,  0.0027,  ..., -0.0068, -0.0107,  0.0086],
        [-0.0129,  0.0141, -0.0004,  ...,  0.0042,  0.0169, -0.0028],
        ...,
        [-0.0032,  0.0101,  0.0003,  ...,  0.0083, -0.0311,  0.0122],
        [ 0.0271,  0.0114,  0.0018,  ..., -0.0354, -0.0179, -0.0135],
        [-0.0176, -0.0081,  0.0029,  ...,  0.0174,  0.0163, -0.0085]],
       device='cuda:0')
reconstructed_weight:tensor([[-0.0041, -0.0152, -0.0017,  ...,  0.0025,  0.0024, -0.0037],
        [ 0.0178, -0.0058,  0.0027,  ..., -0.0068, -0.0107,  0.0086],
        [-0.0130,  0.0141, -0.0003,  ...,  0.0042,  0.0169, -0.0028],
        ...,
        [-0.0032,  0.0101,  0.0003,  ...,  0.0083, -0.0311,  0.0122],
        [ 0.0271,  0.0114,  0.0018,  ..., -0.0353, -0.0179, -0.0135],
        [-0.0176, -0.0081,  0.0029,  ...,  0.0174,  0.0163, -0.0085]],
       device='cuda:0')
tensor(0.1984, device='cuda:0')
tensor([[2074, 1544],
        [2074, 1710],
        [2078, 3238]], device='cuda:0')
tensor(-0.0561, device='cuda:0') tensor(0.1423, device='cuda:0')
tensor(-0.0561, device='cuda:0') tensor(0.1423, device='cuda:0')
tensor(-0.0561, device='cuda:0') tensor(0.1423, device='cuda:0')
  0%|          | 0/32 [00:05<?, ?it/s]

**Comment by horizon86** [comment]
int8? If you use int8, there is a bug that it quantized model using torch.uint8, but save and load it as torch.int8, so just fix it by changing `torch.int8` to `torch.uint8` in `spqr_engine.py`

**Comment by horizon86** [comment]
int4, int3, int2 is correct because all of the int range is inside `int8`
